File: 26. Remove Duplicates from Sorted Array/solution1/solution.cc
#include <vector>

using namespace std;

class Solution {
public:
  int removeDuplicates(vector<int> &nums) {
    int i = 0, j = i + 1;
    int k = 1;
    int n = nums.size();
    while (i < n - 1) {
      // find the next different number
      while (j < n && nums[i] == nums[j])
        j++;
      if (j < n) {
        k++;
        swap(nums[i + 1], nums[j++]);
      }
      i++;
    }
    return k;
  }
};

File: 26. Remove Duplicates from Sorted Array/solution1/Solution.java
class Solution {
  public int removeDuplicates(int[] nums) {
    int l = 1;
    for (int r = 1; r < nums.length; r++) {
      if (nums[r] == nums[l - 1]) {
        continue;
      } else {
        swap(nums, l, r);
        l++;
      }
    }
    return l;
  }
  private void swap(int[] nums, int l, int r) {
    int tmp = nums[l];
    nums[l] = nums[r];
    nums[r] = tmp;
  }
}